File: Assignment_43/Program2.c
//Write a Recursive program which accepts Number from user and return Largest Digit.
//Input :  87983
//Output : 9

#include<stdio.h>

int Max(int iNo)
{
    int iDigit = 0;
    static int iMax = 0;

    if(iNo < 0)
    {
        iNo = -iNo;
    }
    
    if(iNo != 0)
    {
        iDigit = iNo % 10;
        if(iDigit > iMax)
        {
            iMax = iDigit;
        }
        iNo = iNo / 10;

        Max(iNo);
    }
    return iMax;
}

int main()
{
    int iValue = 0,iRet = 0;

    printf("Enter the number : ");
    scanf("%d",&iValue);

    iRet = Max(iValue);

    printf("Largest Digit is : %d\n",iRet);

    return 0;
}

File: Assignment_45/Program2.java
/*Write a program which accept matrix from user and reverse the contents of each row.
Input :   3  2  5  9
          4  3  2  2
          8  4  1  9
          3  9  7  5
Output :  9  5  2  3
          2  2  3  4
          9  1  4  8
          5  7  9  3
*/          
import java.util.*;

class Matrix
{
    private int Arr[][];

    public Matrix(int A,int B)
    {
        Arr = new int [A][B];
    }

    public void Accept()
    {
        System.out.println("Please enter the elements of matrix : ");

        Scanner sobj = new Scanner(System.in);
        int i = 0,j = 0;

        for(i = 0;i < Arr.length;i++)
        {
            for(j = 0;j < Arr[i].length;j++)
            {
                Arr[i][j] = sobj.nextInt();
            }
        }
    }

    public void Display()
    {
        System.out.println("Elements of matrix : ");

        int i = 0,j = 0;

        for(i = 0;i < Arr.length;i++)
        {
            for(j = 0;j < Arr[i].length;j++)
            {
                System.out.print(Arr[i][j]+"\t");
            }
            System.out.println();
        }
    }

    public void ReverseRow()
    {
        int i = 0,j = 0,temp = 0;
        
        for(i = 0;i < Arr.length;i++)
        {
            for(j = 0;j < Arr[i].length/2;j++)
            {
                temp = Arr[i][j];
                Arr[i][j] = Arr[i][Arr[i].length - 1 - j];
                Arr[i][Arr[i].length - 1 - j] = temp;
            }
        }
        System.out.println("Matrix after reversing each row : ");
        Display();
    }
}

class Program2
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number of Rows : ");
        int iRow = sobj.nextInt();

        System.out.println("Enter number of Columns : ");
        int iCol = sobj.nextInt();

        Matrix mobj = new Matrix(iRow,iCol);

        mobj.Accept();

        mobj.Display();

        mobj.ReverseRow();
    }
}